libraries/chain/asset_evaluator.cpp

```diff
@@ -549,7 +554,18 @@ void_result asset_publish_feeds_evaluator::do_apply(const asset_publish_feed_ope
    });
 
    if( !(old_feed == bad.current_feed) )
+   {
+      if( bad.has_settlement() ) // implies head_block_time > HARDFORK_CORE_216_TIME
+      {
+         const auto& mia_dyn = base.dynamic_asset_data_id(d);
+         if( !bad.current_feed.settlement_price.is_null()
+             && ~price::call_price(asset(mia_dyn.current_supply, o.asset_id),
+                                   asset(bad.settlement_fund, bad.options.short_backing_asset),
+                                   bad.current_feed.maintenance_collateral_ratio ) < bad.current_feed.settlement_price )
```

Comment:
are we 100% sure about the `<` at this place? is this correct for any comvination of assetid and short backing asset?